kernel/cmdline.cc

```diff
@@ -14,6 +14,33 @@
 
 struct cmdline_params cmdline_params;
 
+template<typename T>
+struct param_metadata {
+  const char *name;
+  T *pval;
+  const T default_val;
+  void (*on_change)(void);
+};
+
+void dummy() {
+  cprintf("im a %s\n", cmdline_params.disable_pcid ? "dummy" : "smarty");
+  u64 now = nsectime();
+  while (nsectime() < now + 5 * 1e9);
+}
+
+struct param_metadata<bool> binary_params[] = {
+  { "disable_pcid",    &cmdline_params.disable_pcid,    false, dummy },
+  { "keep_retpolines", &cmdline_params.keep_retpolines, false, NULL },
+  { "use_vga",         &cmdline_params.use_vga,         true,  NULL },
+  { "mitigate_mds",    &cmdline_params.mds,             true,  NULL },
+};
```

Comment:
This list is missing "lazy_barrier"

Also, the `on_change` function for "mitigate_mds" "keep_retpolines" and "lazy_barrier" should all be apply_hotpatches from hotpatch.cc (feel free to add a declaration for it in kernel.hh)